code/modules/recycling/disposal.dm

```diff
@@ -1308,46 +1309,41 @@
 	icon_state = "pipe-t"
 	var/obj/linked 	// the linked obj/machinery/disposal or obj/disposaloutlet
 
-/obj/structure/disposalpipe/trunk/New()
-	..()
+/obj/structure/disposalpipe/trunk/Initialize()
+	..() //Lateload below
 	dpdir = dir
-	spawn(1)
-		getlinked()
+	return INITIALIZE_HINT_LATELOAD
 
+/obj/structure/disposalpipe/trunk/LateInitialize()
+	if(!linked)
+		getlinked()
 	update()
-	return
+
+/obj/structure/disposalpipe/trunk/Destroy()
+	if(linked)
+		if(istype(linked, /obj/structure/disposaloutlet))
+			var/obj/structure/disposaloutlet/D = linked
+			D.trunk = null
+		else if(istype(linked, /obj/machinery/disposal))

```

Comment:
```suggestion
		if(istype(linked, /obj/machinery/disposal))

```
Fix compile error since disposaloutlet evidently doesn't link back to us in our codebase.